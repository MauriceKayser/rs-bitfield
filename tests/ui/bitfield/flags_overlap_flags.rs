//! Assertion failure: _FLAGS_IN_FIELD_0_OVERLAP_WITH_FLAGS_IN_FIELD_1

extern crate alloc;

#[bitfield::bitfield(8)]
struct BitField {
    flags: Flags,
    flags2: Flags2
}

#[derive(Copy, Clone, Debug, bitfield::Flags)]
#[repr(u8)]
enum Flags {
    F0,
    F1,
    F3 = 3 // This flag overlaps with `Flags2::G3`.
}

#[derive(Copy, Clone, Debug, bitfield::Flags)]
#[repr(u8)]
enum Flags2 {
    G3 = 3, // This flag overlaps with `Flags::F3`.
    G4,
    G5,
    G7 = 7
}

fn main() {}