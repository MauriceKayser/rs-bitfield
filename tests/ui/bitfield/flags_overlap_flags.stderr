error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/flags_overlap_flags.rs:7:5
  |
7 |     flags: Flags,
  |     ^^^^^ the evaluated program panicked at 'Flags in field "flags" overlap with flags in field "flags2"', $DIR/tests/ui/bitfield/flags_overlap_flags.rs:7:5
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
